### ISSUE TITLE
Fetch multiple elements ($$)

### ISSUE THREAD

**Issue by adrianna-u** [js-soft/wdi5#115]
Hi,

is this possible to select multiple elements in wdi5? (Similar to wdio $$ notation and uiveri5 element.all feature)

We are using following locators in UIVeri5 and looking for a replacement
``` js
 element.all(by.control({
   controlType: 'sap.m.Title'
}))
```
or WDIO
``` js
get allTitles_WDIO() { return $$('.sapMTitle')}

```
Thanks
Adrianna

**Comment by vobu** [comment]
hi, there's no api yet for selecting all controls of a certain type in `wdi5`.
but i'm labelling this as enhancement/feature request - so once we can actually make a public road map (don't ask), it'll be on the dev plan

**Comment by maximnaidenov** [comment]
"I have a dream" -  WDIO custom locator e.g. $ui5({selector:{}}) similar to $react(), then all other things (chaining, methods) will "just  work"

**Comment by vobu** [comment]
> "I have a dream" - WDIO custom locator e.g. $ui5({selector:{}}) similar to $react(), then all other things (chaining, methods) will "just work"

let's see what we can do to fulfill your dreams ☁️ 😺

**Comment by vobu** [comment]
support
- RegEx 
- id wildcard
- controlType
as criteria for a "select all"

eventually "all descendants/ancestors of" ?

**Comment by vobu** [comment]
just to provide an update here @adrianna-u: we've discussed different approaches and are about to prototype some.
there will for sure be an implementation until ~ end of the month (Mar 22), going into the regular `0.9.0` release

**Comment by dominikfeininger** [comment]
https://sapui5.hana.ondemand.com/sdk/#/api/sap.ui.test.RecordReplay%23methods/sap.ui.test.RecordReplay.findAllDOMElementsByControlSelector

**Comment by vobu** [comment]
closed via #212, documented at https://js-soft.github.io/wdi5/#/usage?id=allcontrols and (will be) part of the `rc4`-release!
